src/Model/Post/Media.php

```diff
@@ -616,6 +621,63 @@ public static function getType(string $mimeType): int
 		return $type;
 	}
 
+	/**
+	 * Fetch video dimensions using getID3
+	 *
+	 * @param array $media
+	 * @return array media with added dimensions
+	 */
+	private static function getVideoDimensions(array $media): array
+	{
+		if (isset($media['width']) && isset($media['height']) && $media['width'] > 0 && $media['height'] > 0) {
+			return $media;
+		}
+
+		DI::logger()->debug('Fetch video dimensions', ['uri-id' => $media['uri-id'], 'url' => $media['url']]);
+		$timestamp  = microtime(true);
+		$timeout    = DI::config()->get('system', 'xrd_timeout');
+		$curlResult = DI::httpClient()->get($media['url'], HttpClientAccept::AS_DEFAULT, [HttpClientOptions::TIMEOUT => $timeout, HttpClientOptions::HEADERS => ['Range' => 'bytes=0-1000000']]);
+		if (!$curlResult->isSuccess()) {
+			DI::logger()->notice('Could not fetch video', ['uri-id' => $media['uri-id'], 'url' => $media['url'], 'code' => $curlResult->getReturnCode()]);
+			return $media;
+		}
+
+		if ((!isset($media['modified']) || !$media['modified']) && isset($curlResult->getHeader('Last-Modified')[0]) && $curlResult->getHeader('Last-Modified')[0] != '') {
+			$media['modified'] = DateTimeFormat::utc($curlResult->getHeader('Last-Modified')[0]);
+			if (!isset($media['published']) || !$media['published']) {
+				$media['published'] = $media['modified'];
+			}
+		}
+
+		$video = $curlResult->getBodyString() ?? '';
+		if (!$video) {
+			DI::logger()->notice('Empty video content', ['uri-id' => $media['uri-id'], 'media' => $media]);
+			return $media;
+		}
+
+		$tempfile = tempnam(System::getTempPath(), 'video-');
+		file_put_contents($tempfile, $video);
+		$getID3 = new getID3();
+		$info   = $getID3->analyze($tempfile);
+		unlink($tempfile);
+		$runtime = number_format(microtime(true) - $timestamp, 3);
+
+		if (isset($info['video']['resolution_x']) && isset($info['video']['resolution_y'])) {
+			$media['width']  = $info['video']['resolution_x'];
+			$media['height'] = $info['video']['resolution_y'];
+			DI::logger()->debug('Detected video dimensions', ['runtime' => $runtime, 'uri-id' => $media['uri-id'], 'url' => $media['url'], 'width' => $media['width'], 'height' => $media['height']]);
+		} elseif (isset($info['quicktime']['video']['resolution_x']) && isset($info['quicktime']['video']['resolution_y'])) {
+			$media['width']  = $info['quicktime']['video']['resolution_x'];
+			$media['height'] = $info['quicktime']['video']['resolution_y'];
```

Comment:
In case getID3 doesn't return a valid resolution, at least set it to `0`.

```suggestion
			$media['width']  = (int) $info['quicktime']['video']['resolution_x'] ?? 0;
			$media['height'] = (int) $info['quicktime']['video']['resolution_y'] ?? 0;
```

Reply:
I would prefer it to be passed through when `getID3` returns a null. The value 0 is misleading here. For example, it could be an audio file in a video container (AFAIK, this is possible). In this case, I assume that `width` and `height` will be `0`. However, if there is a problem, these values will be `null`. This means that we can handle these values differently later on.